binaries/coordinator/src/run/custom.rs

```diff
@@ -15,33 +15,42 @@ pub(super) async fn spawn_custom_node(
     communication: &dora_core::config::CommunicationConfig,
     working_dir: &Path,
 ) -> eyre::Result<tokio::task::JoinHandle<eyre::Result<(), eyre::Error>>> {
-    let path = if source_is_url(&node.source) {
+    let mut is_path = source_is_path(&node.source, working_dir);
+
+    // Use path if node.source correspond to a path
+    let mut command = if let Ok(path) = &is_path {
+        tokio::process::Command::new(path)
+    // Use url if node.source is a url
+    } else if source_is_url(&node.source) {
         // try to download the shared library
         let target_path = Path::new("build")
             .join(node_id.to_string())
             .with_extension(EXE_EXTENSION);
         download_file(&node.source, &target_path)
             .await
             .wrap_err("failed to download custom node")?;
-        target_path
+        is_path = Ok(target_path.clone());
+        tokio::process::Command::new(target_path)
+    // Use node.source as a command if it is not in path and is not a url
+    } else if cfg!(target_os = "windows") {
+        let mut cmd = tokio::process::Command::new("cmd");
+        cmd.args(["/C", &node.source]);
+        cmd
     } else {
-        let raw = Path::new(&node.source);
-        if raw.extension().is_none() {
-            raw.with_extension(EXE_EXTENSION)
-        } else {
-            raw.to_owned()
-        }
-    };
-
-    let cmd = working_dir
-        .join(&path)
-        .canonicalize()
-        .wrap_err_with(|| format!("no node exists at `{}`", path.display()))?;
+        let mut cmd = tokio::process::Command::new("sh");
 
-    let mut command = tokio::process::Command::new(cmd);
+        // Argument are passed in the command string.
+        // Arguments passed after string enclosure will be ignored.
+        cmd.args([
+            "-c",
+            &format!("{} {}", node.source, &node.args.clone().unwrap_or_default()),
+        ]);
+        cmd
+    };
```

Comment:
I'm not sure if it is a good idea to treat `source` as a bash/cmd script. It is a bit unexpected because all other uses of `source` just point to an executable without arguments. And how should a source command such as `conda activate env && python node.py` be combined with additional `args`? For example, should the args be passed to the shell, to the first command, to the second command, or to both commands?

How about we add a special `source` named `shell` or `cmd` (or similar) and then put the bash/cmd line into the `args` key? This way, it would be more obvious what happens.

Reply:
That's a very good idea! Let's do `source: shell` to make it less confusing.